lib/wibox/widget/imagebox.lua

```diff
@@ -20,43 +19,45 @@ local math = math
 
 local unpack = unpack or table.unpack -- luacheck: globals unpack (compatibility with Lua 5.1)
 
+-- Safe load for optional Rsvg module
+local Rsvg = nil
+do
+    local success, err = pcall(function() Rsvg = lgi.Rsvg end)
+    if not success then
+        gdebug.print_warning(debug.traceback("Could not load Rsvg: " .. tostring(err)))
+    end
+end
+
 local imagebox = { mt = {} }
 
 ---Load rsvg handle form image file
 ---@tparam string file Path to svg file.
 ---@return Rsvg handle
----@treturn number Image default width
----@treturn number Image default height
-local function try_to_load_svg(file)
-    if not (Rsvg and file:match("%.svg$")) then return end
+local function load_rsvg_handle(file)
+    if not Rsvg then return end
 
     local handle, err = Rsvg.Handle.new_from_file(file)
-    if not handle then
-        gdebug.print_error(debug.traceback(
-            "Rsvg failed to load '" .. tostring(file) .. "': " .. tostring(err)))
-        return
+    if not err then
+        return handle
     end
-
-    local dim = handle:get_dimensions()
-    return handle, dim.width, dim.height
 end
 
 -- Draw an imagebox with the given cairo context in the given geometry.
 function imagebox:draw(_, cr, width, height)
     if width == 0 or height == 0 or not self._private.default then return end
 
+    -- Set the clip
+    if self._private.clip_shape then
+        cr:clip(self._private.clip_shape(cr, width, height, unpack(self._private.clip_args)))
+    end
+
     if not self._private.resize_forbidden then
         -- Let's scale the image so that it fits into (width, height)
         local w, h = self._private.default.width, self._private.default.height
         local aspect = math.min(width / w, height / h)
         cr:scale(aspect, aspect)
     end
 
-    -- Set the clip
-    if self._private.clip_shape then
-        cr:clip(self._private.clip_shape(cr, width, height, unpack(self._private.clip_args)))
-    end
-
```

Comment:
Can you rebase on master? I think that should get rid of this change. Sorry for extra trouble I am causing here.

Reply:
Done.